tests/staticfiles_tests/test_storage.py

```diff
@@ -244,6 +244,56 @@ def test_cache_key_memcache_validation(self):
         self.assertEqual(cache_key, 'staticfiles:821ea71ef36f95b3922a77f7364670e7')
 
 
+@override_settings(
+    **dict(
+        TEST_SETTINGS,
+        STATICFILES_STORAGE=(
+            'staticfiles_tests.storage.MultiCachedStaticFilesStorage'
+        ),
+        DEBUG=False,  # see note above
+    )
+)
+class TestMultiCachedStorage(BaseCollectionTestCase, BaseStaticFilesTestCase,
+        SimpleTestCase):
+
+    def setUp(self):
+        storage.staticfiles_storage.hashed_files.clear()  # avoid cache interference
+        super(TestMultiCachedStorage, self).setUp()
+
+    def cached_file_path(self, path):
+        fullpath = self.render_template(self.static_template_snippet(path))
+        return fullpath.replace(settings.STATIC_URL, '')
+
+    def test_multi_extension_patterns(self):
+        """ See #19670 """
+        self.assertEqual(
+            settings.STATICFILES_STORAGE,
+            'staticfiles_tests.storage.MultiCachedStaticFilesStorage'
+        )
+        self.assertEqual(
+            storage.staticfiles_storage._wrapped.__class__.__name__,
+            'MultiCachedStaticFilesStorage'
+        )
+
+        # Confirm CSS files haven't been touched by JS patterns.
+        relpath = self.cached_file_path("cached/import.css")
+        self.assertEqual(relpath, "cached/import.2b1d40b0bbd4.css")
+        with storage.staticfiles_storage.open(relpath) as relfile:
+            self.assertIn(
+                b"""import url("styles.bb84a0240107.css")""",
+                relfile.read()
+            )
+
+        # Confirm JS patterns have been applied to JS files.
+        relpath = self.cached_file_path("cached/test.js")
+        self.assertEqual(relpath, "cached/test.62789ffcd280.js")
```

Comment:
This is the failing assertion on Windows. I think it might have to do with the file being written with Windows vs. Unix line endings. If you remove the assertion, the rest of the test passes.